program_intent_engagement/apps/api/v1/views.py

```diff
@@ -72,3 +71,78 @@ def post(self, request):
             data = {"detail": "Invalid data", "errors": serializer.errors}
 
         return Response(status=response_status, data=data)
+
+
+class MostRecentAndCertainIntentsView(APIView):
+    """
+    Retrieve a list of user intents
+    This endpoint returns a list of intents specified by the user.
+      Grouped by programs and sorted by most recent
+      and CERTAIN (if available, otherwise MAYBE) intent
+    Path: /api/[version]/intents/most-recent-and-certain
+    Returns:
+     * 200: OK, list of intents
+     * 404: User not found
+    """
+
+    authentication_classes = (JwtAuthentication,)
+    serializer_class = ProgramIntentSerializer
+
+    @classmethod
+    def grab_best_intents(cls, user_intents):
+        """Given a queryset of intents by program_uuid
+          returns a dictionary of best intent per program
+
+        Args:
+            user_intents (QuerySet): A queryset of ProgramIntent Objects
+
+        Returns:
+            Dict: A dictionary with program_uuid as keys and program_intent as values
+        """
+        best_intents_dict = {}
+
+        # sorting priority for certainty
+        certainty_order = {"CERTAIN_YES": 1, "CERTAIN_NO": 1, "MAYBE": 2}
+
+        sorted_intents_list = list(user_intents.values().order_by("program_uuid"))
+
+        for program_uuid, intents in groupby(
```

Comment:
What do you think about making a note that `groupby` requires a list sorted by the key (`program_uuid`)?

Reply:
This is a good idea, I think i might've had it at one point and then deleted it lol.